### PR TITLE
chore: update to lotus 1.5.0-pre1 and actors v3

### PR DESCRIPTION
**Pull request by iand**
Tests for actors v3 are copy/paste of v2 tests with fixups for the types. The main change for v3 is the structure of the HAMT.

### CODE REVIEW

File: tasks/actorstate/reward_test.go [hunk 1]
```diff
@@ -41,7 +43,7 @@ func TestRewardExtractV0(t *testing.T) {
 	stateCid, err := mapi.Store().Put(ctx, state)
 	require.NoError(t, err)
 
-	minerAddr := tutils.NewIDAddr(t, 00)
+	minerAddr := tutils.NewIDAddr(t, 0o0)
```
Comment (placer14) [inline tasks/actorstate/reward_test.go:46]: May be nothing, but curious why this was needed?

**Review by placer14 — APPROVED** [review]
Confused how the `ffi-stub` was building before the types or constants were present. I don't see where it was changed but checking out locally it obviously points to that module. Just some small confusion, but looks good otherwise. 🤝

**Comment by willscott** [comment]
only the subset of ffi stub actually referenced by us or lotus is needed when a module replacement happens.

**Review by frrist — APPROVED** [review]
LGMT! :pray:

**Comment by willscott** [comment]
ℹ️ There is a dependency on statediff for parsing message parameters. that may also need to get pulled up to the v3 dependencies to continue functioning. I expect to do that this week.

File: tasks/actorstate/reward_test.go (hunk 1 above)
Comment (iand) [reply tasks/actorstate/reward_test.go:46]: Not my doing. I think gofumpt does it. Numbers starting with 0 are octal in Go, and they added `0o` as a consistent way to write them to match `0x`

**Comment by iand** [comment]
> Confused how the `ffi-stub` was building before the types or constants were present. I don't see where it was changed but checking out locally it obviously points to that module. Just some small confusion, but looks good otherwise.

They were in the real ffi package but not imported into Lotus until recently.

**Pull request merged**